### PR TITLE
API 14995 clear id option

### PR DESCRIPTION
**Pull request by bschofield-va**
- wip - partially working clear IDs
- Prototype support for clear IDs

### CODE REVIEW

File: vista-fhir-query-tests/src/main/java/gov/va/api/health/vistafhirquery/tests/SystemDefinitions.java [hunk 1]
```diff
@@ -222,6 +228,56 @@ public static SystemDefinition systemDefinition() {
     }
   }
 
+  private TestIds clarifyTestIds(TestIds in) {
```
Comment (jhulbert-va) [inline vista-fhir-query-tests/src/main/java/gov/va/api/health/vistafhirquery/tests/SystemDefinitions.java:231]: Only local?

File: vista-fhir-query-tests/src/main/java/gov/va/api/health/vistafhirquery/tests/TestIds.java [hunk 2]
```diff
@@ -18,8 +18,6 @@
 public final class TestIds {
   @NonNull String appointment;
 
-  @NonNull String coverage;
```
Comment (jhulbert-va) [inline vista-fhir-query-tests/src/main/java/gov/va/api/health/vistafhirquery/tests/TestIds.java:21]: 🚮

**Review by jhulbert-va — COMMENTED** [review]
This will change the current IDs in the wild won't it? Should we consult CPM?

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4SiteAppointmentController.java [hunk 3]
```diff
@@ -62,6 +64,7 @@ public Appointment appointmentRead(
       @PathVariable("site") String site,
       @PathVariable("id") String id) {
     SegmentedVistaIdentifier ids = parseOrDie(id);
+    log.info("Read {}", ids);
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4SiteAppointmentController.java:67]: Do we want to be logging this? My feeling would be no.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/witnessprotection/WitnessProtectionAdvice.java [hunk 4]
```diff
@@ -316,6 +345,72 @@ public <R extends Resource> String toPublicId(Class<R> resourceType, String priv
             () -> new IllegalStateException("Could not generate private id for resource."));
   }
 
+  /**
+   * OMG I HATE YOU SO MUCH. THIS IS A DUMB HACK TO HANDLE THE TRANSITION CASE BETWEEN SUPPORTING
+   * ENCODED AND CLEAR IDS. THIS WILL BE DELETED ONCE WE FULLY TRANSITION TO CLEAR.
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/witnessprotection/WitnessProtectionAdvice.java:350]: Task?

File: vista-fhir-query-tests/src/main/java/gov/va/api/health/vistafhirquery/tests/SystemDefinitions.java (hunk 1 above)
Comment (bschofield-va) [reply vista-fhir-query-tests/src/main/java/gov/va/api/health/vistafhirquery/tests/SystemDefinitions.java:231]: yes for now ... I'm not 100% confident that this works in every case.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medicationdispense/MedicationDispenseId.java [hunk 5]
```diff
@@ -30,7 +40,7 @@ public static MedicationDispenseId fromString(@NonNull String privateId) {
 
   @Override
   public String toString() {
-    return vistaId().pack() + ":" + fillDate();
+    return vistaId().pack() + "." + fillDate();
```
Comment (jhulbert-va) [inline vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medicationdispense/MedicationDispenseId.java:43]: Should this be a `-`? Some of our IDs have fileman dates in them, right?

Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/medicationdispense/MedicationDispenseId.java:43]: ^ these do not appear too

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/witnessprotection/WitnessProtectionAdvice.java (hunk 4 above)
Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/witnessprotection/WitnessProtectionAdvice.java:350]: when ID encoding is removed from VFQ, I envision all of this going away... like the entire class hierarchy involved in WitnessProtection. I don't really feel the need to mark my complaints about how dumb this hack is with a Jira item.

File: vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4SiteAppointmentController.java (hunk 3 above)
Comment (bschofield-va) [reply vista-fhir-query/src/main/java/gov/va/api/health/vistafhirquery/service/controller/appointment/R4SiteAppointmentController.java:67]: I guess not... though the IDs show up in our logs in other places.

**Review by jhulbert-va — DISMISSED** [review]

**Review by eclendenning-va — APPROVED** [review]

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**